scripts/variantstore/wdl/extract/summarize_task_monitor_logs_from_file.py

```diff
@@ -0,0 +1,204 @@
+            shard = tokens[index][6:]
+            task = tokens[index - 1][5:]  # Strip off the 'call-' prefix
+        else:
+            task = tokens[index][5:]
+
+        summary = f"{total_memory}\t{MaxMem}\t{MaxMemPct}\t{total_disk}\t{MaxDisk}\t{MaxDiskPct}\t{task}\t{shard}\t" \
+                  f"{os.path.abspath(mlog_file)}\n"
+        output.write(summary)
+
+
+def parse_cpu_usage_line(line):
+    p = "^\\* CPU usage\\: (\\d+\\.\\d+)%$"  # Looks Like: * CPU usage: 17.6%
+    m = re.match(p, line)
+    if m is not None:
+        cpu = float(m.group(1))
+        global MaxCpu
+        if cpu > MaxCpu:
+            MaxCpu = cpu
+    else:
+        # Check if it's a nan (we see this sometimes at startup)
+        p2 = "^\\* CPU usage\\: -nan\\%$"  # * CPU usage: -nan%
+        m2 = re.match(p2, line)
+        if m2 is None:
+            # Check if it's just empty
+            p3 = "^\\* CPU usage\\: *$"  # * CPU usage:
+            m3 = re.match(p3, line)
+            if m3 is None:
+                eprint(f"ERROR: Line '{line}' does not look like a CPU usage line. Is this a monitoring_log file?")
+                sys.exit(1)
+
+
+def parse_memory_usage_line(line):
+    p = "^\\* Memory usage\\: (\\d+\\.\\d+) \\S+ (\\d+(\\.\\d*)?)\\%$"  # Looks Like: * Memory usage: 1.79 GiB 1.8%
+    m = re.match(p, line)
+    if m is None:
+        eprint(f"ERROR: Line '{line}' does not look like a Memory usage line. Is this a monitoring_log file?")
+        sys.exit(1)
+    mem = float(m.group(1))
+    mem_pct = float(m.group(2))
+    global MaxMem
+    global MaxMemPct
+    if mem > MaxMem:
+        MaxMem = mem
+        MaxMemPct = mem_pct
+
+
+def parse_disk_usage_line(line):
+    p = "^\\* Disk usage\\: (\\d+\\.\\d+) \\S+ (\\d+(\\.\\d*)?)\\%$"  # Looks Like: * Disk usage: 22.000 GiB 3%
+    m = re.match(p, line)
+    if m is None:
+        eprint(f"ERROR: Line '{line}' does not look like a Disk usage line. Is this a monitoring_log file?")
+        sys.exit(1)
+    disk = float(m.group(1))
+    disk_pct = float(m.group(2))
+    global MaxDisk
+    global MaxDiskPct
+    if disk > MaxDisk:
+        MaxDisk = disk
+        MaxDiskPct = disk_pct
+
+
+def advance_to_section(fp, section_header_start):
+    """
+    A method to advance to the runtime block in the file
+    :param fp: The file's file pointer
+    :param section_header_start: The (starting part) of the section header in the file.
+    :return: The nextmost line that is not a header line
+    """
+    while line := fp.readline():
+        if line.startswith(section_header_start):
+            return line
+
+
+def eprint(*the_args, **kwargs):
+    print(*the_args, file=sys.stderr, **kwargs)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(allow_abbrev=False,
+                                     description='A tool to summarize the output of multiple monitoring logs')
+
+    parser.add_argument('--input', type=str, help='Monitoring log FOFN', required=True)
```

Comment:
Would it be much work to modify the existing `summarize_task_monitor_logs.py` script to take either the FOFN or the list of inputs?

Reply:
It's not a lot of work; my thinking was that creating a new script (or modifying the existing one) was already scope creep from the ticket at hand (basically "make sure we update the docs on how to create a PGEN deliverable for AoU"), so I didn't want it to creep further.